src/highdicom/ann/content.py

```diff
@@ -496,6 +498,73 @@ def primary_anatomic_structures(self) -> List[CodedConcept]:
             return []
         return list(self.PrimaryAnatomicStructureSequence)
 
+    def get_graphic_data(
+        self,
+        coordinate_type: Union[str, AnnotationCoordinateTypeValues]
+    ) -> List[np.ndarray]:
+        """Get spatial coordinates of all graphical annotations.
+
+        Parameters
+        ----------
+        coordinate_type: Union[str, highdicom.ann.AnnotationCoordinateTypeValues]
+            Coordinate type of annotation
+
+        Returns
+        -------
+        List[numpy.ndarray]
+            Two-dimensional array of floating-point values representing either
+            2D or 3D spatial coordinates for each graphical annotation
+
+        """  # noqa: E501
+        coordinate_type = AnnotationCoordinateTypeValues(coordinate_type)
+        if self._graphic_data is None:
```

Comment:
Technically, this caching mechanism doesn't take into account the value of the `coordinate_type` parameter.

Consequently, subsequent calls to the method can pass the wrong value and it will return the old data. I doubt this would happen much in practice, but still seems like a potentially nasty bug waiting to happen:

```python
data = group.get_graphic_data('SCOORD')
...
more_data = group.get_graphic_data('SCOORD3D')  # will return data, which is definitely not the expected behaviour...
```

I suggest you consider caching the coordinate type too somehow between calls.

EDIT this is especially pernicious when you realise that users may call this function indirectly through the `get_coordinates` method on multiple occasions and potentially pass different `coordinate_type`s (not that there's any reason to of course, but it still *could* happen...).

Reply:
Addressed via c61aa724b733463e8e3cad9275a28b2f9fbf288a